src/libfetchers/path.cc

```diff
@@ -140,6 +142,14 @@ struct PathInputScheme : InputScheme
                 mtime = dumpPathAndGetMtime(absPath.string(), sink, defaultPathFilter);
             });
             storePath = store->addToStoreFromDump(*src, "source");
```

Comment:
I think we should actually have the cache populated or by-passed in both cases....